x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/time_scaling.tsx

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiLink,
+  EuiFormRow,
+  EuiSelect,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiButtonIcon,
+  EuiText,
+  EuiPopover,
+  EuiButtonEmpty,
+  EuiSpacer,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useState } from 'react';
+import { StateSetter } from '../../types';
+import { IndexPatternColumn, operationDefinitionMap } from '../operations';
+import { mergeLayer } from '../state_helpers';
+import { unitSuffixesLong } from '../suffix_formatter';
+import { TimeScaleUnit } from '../time_scale';
+import { IndexPatternPrivateState } from '../types';
+
+const DEFAULT_TIME_SCALE = 'm' as const;
+
+export function TimeScaling({
+  selectedColumn,
+  columnId,
+  layerId,
+  state,
+  setState,
+}: {
+  selectedColumn: IndexPatternColumn;
+  columnId: string;
+  layerId: string;
+  state: IndexPatternPrivateState;
+  setState: StateSetter<IndexPatternPrivateState>;
+}) {
+  const [popoverOpen, setPopoverOpen] = useState(false);
+  const layer = state.layers[layerId];
+  const hasDateHistogram = layer.columnOrder.some(
+    (colId) => layer.columns[colId].operationType === 'date_histogram'
+  );
+  const selectedOperation = operationDefinitionMap[selectedColumn.operationType];
+  if (
+    !selectedOperation.timeScalingMode ||
+    selectedOperation.timeScalingMode === 'disabled' ||
+    !hasDateHistogram
+  ) {
+    return null;
+  }
+
+  if (!selectedColumn.timeScale) {
+    return (
+      <EuiText textAlign="right">
+        <EuiSpacer size="s" />
+        <EuiPopover
+          ownFocus
+          button={
+            <EuiButtonEmpty
+              size="xs"
+              iconType="arrowDown"
+              iconSide="right"
+              data-test-subj="indexPattern-time-scaling-popover"
+              onClick={() => {
+                setPopoverOpen(true);
+              }}
+            >
+              {i18n.translate('xpack.lens.indexPattern.timeScale.advancedSettings', {
+                defaultMessage: 'Add advanced options',
+              })}
+            </EuiButtonEmpty>
+          }
+          isOpen={popoverOpen}
+          closePopover={() => {
+            setPopoverOpen(false);
+          }}
+        >
+          <EuiText size="s">
+            <EuiLink
+              data-test-subj="indexPattern-time-scaling-enable"
+              color="text"
```

Comment:
I followed the mockups here:
<img width="222" alt="Screenshot 2020-11-24 at 11 59 46" src="https://user-images.githubusercontent.com/1508364/100085640-93faa000-2e4c-11eb-823b-4df405b37b63.png">

IMHO this makes sense because of the similarity to the official context menu component:
<img width="354" alt="Screenshot 2020-11-24 at 12 00 21" src="https://user-images.githubusercontent.com/1508364/100085724-af65ab00-2e4c-11eb-9c53-4df10d12660b.png">

Reply:
I won't block this any more, but I do think this is a confusing design because it lacks the affordances of the EUI examples. The main reason I think this is problematic is that it's using a menu-oriented design, but only has one item.